autotest/utilities/test_ogr2ogr.py

```diff
@@ -2879,3 +2880,196 @@ def test_ogr2ogr_check_identity_transformation(ogr2ogr_path, x, y, srid):
+    dest_file_name = tempfile.mktemp(dest_ext, dest_file_base_name)
+
+    dr = ogr.GetDriverByName(src_driver)
+    src_width_includes_sign = (
+        dr.GetMetadata_Dict().get("DMD_NUMERIC_FIELD_WIDTH_INCLUDES_SIGN", "NO")
+        == "YES"
+    )
+    src_width_includes_point = (
+        dr.GetMetadata_Dict().get(
+            "DMD_NUMERIC_FIELD_WIDTH_INCLUDES_DECIMAL_SEPARATOR", "NO"
+        )
+        == "YES"
+    )
+
+    assert src_width_includes_sign == exp_src_minus_flag
+    assert src_width_includes_point == exp_src_decimal_flag
+
+    # Assume width=5 includes sign and point to store 12.34 or -1.23 (but not -12.34)
+    # For SQL it would be precision=4, scale=2
+    width = 5
+    precision = 2
+
+    # If the source does not include decimal separator, we can reduce width by one
+    if not src_width_includes_point:
+        width -= 1
+
+    ds = dr.CreateDataSource(src_file_name)
+
+    # For CSV we want csvt to support width and precision
+    options = []
+    if src_driver == "CSV":
+        options.append("CREATE_CSVT=YES")
+        options.append("GEOMETRY=AS_WKT")
+
+    lyr = ds.CreateLayer(src_file_base_name, geom_type=ogr.wkbPoint, options=options)
+    field_def = ogr.FieldDefn("num_5_2", ogr.OFTReal)
+    field_def.SetWidth(width)
+    field_def.SetPrecision(precision)
+    lyr.CreateField(field_def)
+
+    lyr_def = lyr.GetLayerDefn()
+    field_idx = lyr_def.GetFieldIndex("num_5_2")
+    f = ogr.Feature(lyr_def)
+    f.SetField(field_idx, 12.34)
+    f.SetGeometry(ogr.CreateGeometryFromWkt("POINT(0 1)"))
+    lyr.CreateFeature(f)
+
+    f = ogr.Feature(lyr_def)
+    f.SetField(field_idx, -1.23)
+    f.SetGeometry(ogr.CreateGeometryFromWkt("POINT(0 1)"))
+    lyr.CreateFeature(f)
+
+    # This does not fit if minus sign is included in width
+    f = ogr.Feature(lyr_def)
+    f.SetField(field_idx, -12.34)
+    f.SetGeometry(ogr.CreateGeometryFromWkt("POINT(0 1)"))
+    lyr.CreateFeature(f)
+    f = None
+    lyr = None
+    ds = None
+
+    # Verify source
+    ds = ogr.Open(src_file_name)
+    lyr = ds.GetLayer(0)
+    assert lyr is not None
+    lyr_def = lyr.GetLayerDefn()
+    field_idx = lyr_def.GetFieldIndex("num_5_2")
+    field_def = lyr_def.GetFieldDefn(field_idx)
+    assert field_def.GetWidth() == width
+    assert field_def.GetPrecision() == precision
+
+    features = [f for f in lyr]
+    f = features[0]
+    assert f.GetFieldAsDouble(field_idx) == 12.34
+    f = features[1]
+    assert f.GetFieldAsDouble(field_idx) == -1.23
+    f = features[2]
+
+    if src_driver in PRECISION_UNRELIABLE_DRIVERS:
+        assert f.GetFieldAsDouble(field_idx) == -12.34
+    else:
+        assert f.GetFieldAsDouble(field_idx) == -12.3
+
+    # For CSV we want csvt to support width and precision
+    lco_options = ""
+    if dest_driver == "CSV":
+        lco_options = "-lco CREATE_CSVT=YES"
+
+    # Source is ok, convert with default options
+    gdaltest.runexternal(
```

Comment:
I would move that test to test_ogr2ogr_lib.py, where you can use gdal.VectorTranslate() instead of running an external process.   test_ogr2ogr should be considered a bit legacy now, or to test the binary specific aspects of ogr2ogr (here this functionality testable by the GDALVectorTranslate() functions)